flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -1378,11 +1389,11 @@ private void declineCheckpoint(
     }
 
     public void notifyCheckpointComplete(final long checkpointID) {
-        final AbstractInvokable invokable = this.invokable;
+        final TaskInvokable invokable = this.invokable;
 
-        if (executionState == ExecutionState.RUNNING && invokable != null) {
+        if (executionState == ExecutionState.RUNNING && invokable instanceof CheckpointableTask) {
```

Comment:
as I understand, it is impossible/unexpected that `invokable instanceof CheckpointableTask` would be false. So maybe is it better to replace the `if condition` with `checkArgument`?(The same question to others `instanceof`)